src/iocore/net/SSLNetVConnection.cc

```diff
@@ -1581,28 +1465,18 @@ SSLNetVConnection::sslClientHandShakeEvent(int &err)
       }
     }
 
-    sslHandshakeHookState = HANDSHAKE_HOOKS_OUTBOUND_PRE;
+    this->set_handshake_hook_state(TLSEventSupport::SSLHandshakeHookState::HANDSHAKE_HOOKS_OUTBOUND_PRE);
   }
 
   // Do outbound hook processing here
   // Continue on if we are in the invoked state.  The hook has not yet reenabled
-  if (sslHandshakeHookState == HANDSHAKE_HOOKS_OUTBOUND_PRE_INVOKE) {
+  if (this->is_invoked_state()) {
```

Comment:
is it relevant in which particular `INVOKE` state it is?

Reply:
No, not really in terms of the flow. Like the function name suggests, whether it's one of `INVOKE`s is the key.